libs/base/Control/Function.idr

```diff
@@ -6,5 +6,5 @@ interface Injective (f : a -> b) where
   injective : {x, y : a} -> f x = f y -> x = y
 
 public export
-inj : (f : a -> b) -> Injective f => {x, y : a} -> f x = f y -> x = y
-inj _ = injective
+inj : (0 f : a -> b) -> Injective f => {0 x, y : a} -> (0 _ : f x = f y) -> x = y
```

Comment:
```suggestion
inj : (0 f : a -> b) -> {auto 0 _ : Injective f} -> {0 x, y : a} -> (0 _ : f x = f y) -> x = y
```

Given that we have implementations like `{x : a} -> Injective (x :::) where`, it does make a difference to have `Injective` at 0.